nodejs/eks/nodegroup.ts

```diff
@@ -720,49 +720,38 @@ function createNodeGroupInternal(
             return getOperatingSystem(amiType, operatingSystem);
         });
 
+    const serviceCidr = getClusterServiceCidr(core.cluster.kubernetesNetworkConfig);
+    const clusterMetadata = {
+        name: eksCluster.name,
+        apiServerEndpoint: eksCluster.endpoint,
+        certificateAuthority: eksCluster.certificateAuthority.data,
+        serviceCidr,
+    };
+
     const userdata = pulumi
         .all([
             awsRegion,
-            eksCluster.name,
-            eksCluster.endpoint,
-            eksCluster.certificateAuthority,
+            clusterMetadata,
             cfnStackName,
             args.nodeUserData,
             args.nodeUserDataOverride,
             os,
         ])
-        .apply(
-            ([
-                region,
-                clusterName,
-                clusterEndpoint,
-                clusterCa,
-                stackName,
-                extraUserData,
+        .apply(([region, clusterMetadata, stackName, extraUserData, userDataOverride, os]) => {
+            const userDataArgs: SelfManagedV1NodeUserDataArgs = {
+                nodeGroupType: "self-managed-v1",
+                awsRegion: region.name,
+                kubeletExtraArgs: args.kubeletExtraArgs,
+                bootstrapExtraArgs: args.bootstrapExtraArgs,
+                labels: args.labels,
+                taints: args.taints,
                 userDataOverride,
-                os,
-            ]) => {
-                const userDataArgs: SelfManagedV1NodeUserDataArgs = {
-                    nodeGroupType: "self-managed-v1",
-                    awsRegion: region.name,
-                    kubeletExtraArgs: args.kubeletExtraArgs,
-                    bootstrapExtraArgs: args.bootstrapExtraArgs,
-                    labels: args.labels,
-                    taints: args.taints,
-                    userDataOverride,
-                    extraUserData,
-                    stackName,
-                };
-
-                const clusterMetadata = {
-                    name: clusterName,
-                    apiServerEndpoint: clusterEndpoint,
-                    certificateAuthority: clusterCa.data,
-                };
+                extraUserData,
+                stackName,
+            };
 
-                return createUserData(os, clusterMetadata, userDataArgs);
-            },
-        );
+            return createUserData(os, clusterMetadata, userDataArgs);
```

Comment:
Is clusterMetadata here type-checked or is it <any>? 

Magic.

I see

```
export declare function all<T1, T2, T3, T4, T5, T6, T7, T8>(values: [Input<T1>, Input<T2>, Input<T3>, Input<T4>, Input<T5>, Input<T6>, Input<T7>, Input<T8>]): Output<[Unwrap<T1>, Unwrap<T2>, Unwrap<T3>, Unwrap<T4>, Unwrap<T5>, Unwrap<T6>, Unwrap<T7>, Unwrap<T8>]>;

```
It might actually type-check.

Reply:
Yeah it's type checked :)